cartographer/io/pcd_writing_points_processor.cc

```diff
@@ -32,20 +32,25 @@ namespace {
 
 // Writes the PCD header claiming 'num_points' will follow it into
 // 'output_file'.
-void WriteBinaryPcdHeader(const bool has_color, const int64 num_points,
-                          FileWriter* const file_writer) {
+void WriteBinaryPcdHeader(const bool has_color, const bool has_intensities,
+                          const int64 num_points, FileWriter* const file_writer) {
   std::string color_header_field = !has_color ? "" : " rgb";
   std::string color_header_type = !has_color ? "" : " U";
   std::string color_header_size = !has_color ? "" : " 4";
   std::string color_header_count = !has_color ? "" : " 1";
 
+  std::string intensity_header_field = !has_intensities ? "" : " intensity";
```

Comment:
Is this style of an intensity field commonly used in PCD files? Is it documented somewhere? https://pointclouds.org/documentation/tutorials/pcd_file_format.html does not mention it.

Reply:
Yes, not all possible fields are listed on the site. I usually define points of type pcl::PointXYZI, and these fields are also available in pcd.

For example, you can find all supported fields for the point cloud library file format here https://libpointmatcher.readthedocs.io/en/latest/ImportExport/

Reply:
Thanks.